tests/conftest.py

```diff
@@ -1,20 +1,72 @@
 import datetime
+from collections.abc import Generator
+from typing import Callable
+from unittest.mock import MagicMock, patch
 from uuid import uuid4
 
 import pytest
 from openai.types import CompletionUsage
 from openai.types.chat import ChatCompletionMessage
 from openai.types.chat.chat_completion import ChatCompletion, Choice
 
+from galileo.config import GalileoConfig
 from galileo.resources.models import DatasetContent, DatasetRow, DatasetRowValuesDict
+from galileo_core.constants.request_method import RequestMethod
+from galileo_core.constants.routes import Routes as CoreRoutes
+from galileo_core.schemas.core.user import User
+from galileo_core.schemas.core.user_role import UserRole
 from galileo_core.schemas.protect.rule import Rule, RuleOperator
 from galileo_core.schemas.protect.ruleset import Ruleset
 from tests.testutils.setup import setup_thread_pool_request_capture
 
 
+@pytest.fixture
+def mock_healthcheck(mock_request: Callable) -> Generator[None, None, None]:
+    route = mock_request(method=RequestMethod.GET, path=CoreRoutes.healthcheck)
+    yield
+    assert route.called
+
+
+@pytest.fixture
+def mock_get_current_user(mock_request: Callable) -> Generator[None, None, None]:
+    route = mock_request(
+        RequestMethod.GET,
+        CoreRoutes.current_user,
+        json=User.model_validate({"id": uuid4(), "email": "user@example.com", "role": UserRole.user}).model_dump(
+            mode="json"
+        ),
+    )
+    yield
+    assert route.called
+
+
+@pytest.fixture
+def mock_login_api_key(mock_request: Callable) -> Generator[None, None, None]:
+    route = mock_request(RequestMethod.POST, CoreRoutes.api_key_login, json={"access_token": "secret_jwt_token"})
+    yield
+    assert route.called
+
+
+@pytest.fixture
+def mock_decode_jwt() -> Generator[MagicMock, None, None]:
+    with patch("galileo_core.schemas.base_config.jwt_decode") as _fixture:
+        _fixture.return_value = dict(exp=float("inf"))
+        yield _fixture
+
+
+@pytest.fixture(autouse=True)
+def set_validated_config(
+    mock_healthcheck: None, mock_login_api_key: None, mock_get_current_user: None, mock_decode_jwt: MagicMock
+) -> Generator[None, None, None]:
+    config = GalileoConfig.get()
+    yield
+    config.reset()
+
+
 @pytest.fixture(autouse=True)
 def env_setup(monkeypatch):
     monkeypatch.setenv("OPENAI_API_KEY", "local")
+    monkeypatch.setenv("GALILEO_API_KEY", "api-1234567890")
```

Comment:
We shouldn't need these, no? They are only for authenticated requests.

Either way, if we need env vars for _all_ tests, they should be in `pyproject.toml` under the `pytest` config, like this: https://github.com/rungalileo/api/blob/main/pyproject.toml#L105-L141